e2e/testdata.go

```diff
@@ -61,3 +41,56 @@ func GetSamplePod(name, pvcName string) *k8sv1.Pod {
 	}
 	return pod
 }
+
+// GetSampleVolumeSnapshot creates and returns the VolumeSnapshot for the provided PVC.
+func GetSampleVolumeSnapshot(name string, storageClass string) *snapapi.VolumeSnapshot {
+	vs := &snapapi.VolumeSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			// name = name of pvc + snapshot
+			Name:      name + "snapshot",
+			Namespace: TestNamespace,
+		},
+		Spec: snapapi.VolumeSnapshotSpec{
+			VolumeSnapshotClassName: &storageClass,
+			Source: snapapi.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &name,
+			},
+		},
+	}
+	return vs
+}
+
+// GetSamplePvc returns restore or clone of the pvc based on the kind(kindName) provided.
+func GetSamplePvc(size, name string, volumemode k8sv1.PersistentVolumeMode, storageClass string, sourceType string) *k8sv1.PersistentVolumeClaim {
+	var kindName string
+	if sourceType == "clone" {
```

Comment:
"clone" and "restore" strings are used a multiple places. They should be moved out in a variable and that variable name should be used.